ColorCoding.h

```diff
@@ -0,0 +1,18 @@
+#include <stdio.h>
+
+enum MajorColor {WHITE, RED, BLACK, YELLOW, VIOLET};
+enum MinorColor {BLUE, ORANGE, GREEN, BROWN, SLATE};
+
+typedef struct {
+    enum MajorColor majorColor;
+    enum MinorColor minorColor;
+} ColorPair;
+
+const char* MajorColorNames[];
+const char* MinorColorNames[];
+
+void ColorPairToString(const ColorPair* colorPair, char* buffer);
+void testNumberToPair(int pairNumber, enum MajorColor expectedMajor, enum MinorColor expectedMinor);
```

Comment:
production code and test code should be kept separately, so that test code can be excluded during software shipment to field. It also helps in maintenance.